File: Programming Syntax/C/Basic/loop.c
#include <stdio.h>
#include <string.h>

int main(int argc, char **argv){
	int i = 0;
	for (i=0;i<10;i++){
		printf("Loop 0 iteration %i\n",i);
	}
	
	i=0;
	while (i<10){
		printf("Loop 1 iteration %i\n",i);
		i++;
	}
	
	i=0;
	do{
		printf("Loop 2 iteration %i\n",i);
		i++;
	}while (i<10);
	
	/*breaks*/
	
	i=0;
	while(1){
		if(i==10){
			break;
		}
		printf("Loop 3 iteration %i\n",i);
		i++;
	}
	
	i=0;
	while(1){
		printf("Loop 4 iteration %i\n",i);
		i++;
		if(i<10){
			continue;
		}
		break;
	}	 
	return 0;
}

File: Programming Syntax/Java/Basic/loop.java
public class loop {
  public static void main(String[] args) throws Exception {
    for(int i=0; i<10;i++){
      System.out.println("Loop 0 iteration " + i);
    }
    for(int i=0,j=1; (i<10) && (j<1024); i++,j=j*2){ //you can iterate multiple variables in one loop
      System.out.println("Loop 1 iteration " + i + " - " + j);
    }
    
    Integer[] myArray = new Integer[3]; 
    myArray[0]=0; //assiging values to avoid null pointer exception
    myArray[1]=0;
    myArray[2]=0;
    int i = 0;
    for (int j: myArray){ //this will iterate over a list.
      j = i*i;
      System.out.println("Loop 2 iteration " + i + " - " + j);
      i++;
    }
    
    i = 0;
    while (true) { //while loop
      if (i==10){
        break;
      }
      System.out.println("Loop 3 iteration " + i);
      i++;
    }
    
    i = 0;
    do{ //do-while loop
      System.out.println("Loop 4 iteration " + i);
      i++;
    } while(i<10);
  }
}